clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -666,12 +766,16 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
     }
 
     private void maybeThrowInvalidGroupIdException() {
-        if (!groupId.isPresent() || groupId.get().isEmpty()) {
-            throw new InvalidGroupIdException("To use the group management or offset commit APIs, you must " +
-                    "provide a valid " + ConsumerConfig.GROUP_ID_CONFIG + " in the consumer configuration.");
+        if (!groupMetadata.isPresent()) {
+            throwInInvalidGroupIdException();
         }
     }
 
+    private void throwInInvalidGroupIdException() {
```

Comment:
I find the name `throwInInvalidGroupIdException` a bit strange. `throwInvalidGroupIdException` seems more usual.

Reply:
Yeah, that was already fixed in https://github.com/apache/kafka/pull/14872

Reply:
What a tangled web we weave.